Remove prometheus config from metrics checklist

jordojordo opened the pull request:
Fix #548 

This removes the check for `additionalServiceMonitors` added to the Prometheus config and instead checks for the existence of a `ServiceMonitor` that has a selector pointing to the `kubewarden-controller`. The new step for service monitors will _not_ block the checklist from succeeding.

Also changed the colors of the checkmarks to be green.

![1](https://github.com/rancher/kubewarden-ui/assets/40806497/7c8c7c6b-fdd0-4275-85eb-a27b94ace5c4)

viccuad reviewed (approved):
LGTM from the functionality side (but haven't tested it)

kravciak commented:
I see some potential issues here:
- service monitor [doc link](https://docs.kubewarden.io/operator-manual/telemetry/metrics/quickstart) points to kubewarden, so examples have `kubewarden` namespace, not `cattle-kubewarden-system`, we should point to [UI doc](https://docs.kubewarden.io/operator-manual/ui-extension/metrics#install) , which recommends to install rancher-managed monitoring?
- shouldn't we check for kubewarden-policy-server-default? Is kubewarden-controller service monitor used for something?
- maybe we could add button redirecting to one of those: (would feel more UI-ish) :)
  -  Menu->Monitoring->[Service Monitors](https://172.18.0.2.nip.io/dashboard/c/local/monitoring/monitor#monitoring.coreos.com.servicemonitor) or
  - Menu->More Resources->[Service Monitors](https://172.18.0.2.nip.io/dashboard/c/local/explorer/monitoring.coreos.com.servicemonitor)?

- we should not show grafana before this check is complete, otherwise data are empty

jordojordo commented:
>Is kubewarden-controller service monitor used for something? I see no change when I remove it. Maybe we could check for any service monitor with namespaceSelector === <cattle-kubewarden-system> namespace.

I see no difference when the service monitor is created and then removed, but if the service monitor is never created prometheus does not display any data. I think checking for any service monitor would suffice for the time being.

>we should not show grafana before this check is complete, otherwise data are empty

This check was left as non-blocking due to internal discussion, we should re-discuss this subject.

jordojordo commented:
After some internal discussion, it appears that only a `ServiceMonitor` per Policy Server is necessary for Prometheus to fetch data. So I have updated the PR to now search for a ServiceMonitor tied to the PolicyServer that is either being viewed or that is connected to the policy being viewed.

The checklist is searching all of the fetched ServiceMonitors based the `kubewarden-controller` namespace, and two `matchLabels`:
- `kubewarden/policy-server=<policy-server-id>`
- `app=<policy-server-id`

The label `kubewarden/policy-server` is added automatically to any PolicyServer deployment when a PS is created.

If the service monitor is not found, the checklist will appear as this is necessary for the metrics to contain any data. There is now a button within the ServiceMonitor step of the checklist to add the necessary Service Monitor.


https://github.com/rancher/kubewarden-ui/assets/40806497/59f0910d-d520-4936-83a1-cb57b61b4710

kravciak commented:
matchLabels section is wrong, data are empty
```yaml
# we habe
    selector:
      matchLabels:
        kubewarden/policy-server: default

# this works
    selector:
      matchLabels:
        app: kubewarden-policy-server-default
```

I  think serviceMonitor matchLabel have to match policy-server service, for default one it's:
```
k describe svc -n cattle-kubewarden-system policy-server-default                                                                                                                                                   0.425s
Name:              policy-server-default
Namespace:         cattle-kubewarden-system
Labels:            app=kubewarden-policy-server-default
```

kravciak commented on pkg/kubewarden/l10n/en-us.yaml:
```diff
@@ -149,29 +149,30 @@ kubewarden:
         edit: Edit Config
-        chartError: The Rancher Monitoring chart cannot be found, ensure the repository exists and try reloading the page.
+      serviceMonitor:
+        label: A Service Monitor for the Kubewarden Controller must be created. Follow the <a href="https://docs.kubewarden.io/operator-manual/ui-extension/metrics#install" target="_blank" rel="noopener noreferrer nofollow">documentation</a> to create the additional Service Monitors.
```
```suggestion
        label: A Service Monitor for this Policy Server does not exist. Follow the <a href="https://docs.kubewarden.io/operator-manual/ui-extension/metrics#install" target="_blank" rel="noopener noreferrer nofollow">documentation</a> to learn more about Service Monitors.
```

kravciak commented:
Can we also make `Edit Config` enablement depend on OTEL? Enabling metrics without OTEL fails:
```
helm upgrade --history-max=5 --install=true --namespace=cattle-kubewarden-system --timeout=10m0s --values=/home/shell/helm/values-kubewarden-controller-2.0.3.yaml --version=2.0.3 --wait=true rancher-kubewarden-controller /home/shell/helm/kubewarden-controller-2.0.3.tgz
Error: UPGRADE FAILED: resource mapping not found for name: "kubewarden" namespace: "cattle-kubewarden-system" from "": no matches for kind "OpenTelemetryCollector" in version "opentelemetry.io/v1alpha1"
2023-11-21T08:09:00.679029215Z ensure CRDs are installed first
```

jordojordo commented:
I've fixed the ServiceMonitor check/creation for `matchLabels` to be `app=kubewarden-policy-server-<policy-server-id>`. Also fixed the condition to the "Edit Config" button to be gated by the existence of the opentelemetry service.

jordojordo commented:
e2e will pass when https://github.com/rancher/kubewarden-ui/pull/551 is merged.
